Selenium_Basics2/UnitTest1.cs

```diff
@@ -0,0 +1,161 @@
+            });
+            Assert.IsTrue(searchPanelDisplayedState);
+
+            var searchPanelClick = _chrome.FindElement(By.XPath("//*[@class='header-search__input frequent-searches__input']"));
+            searchPanelClick.Click();
+            searchPanelClick.SendKeys("Automation");
+
+            var searchClick = _chrome.FindElement(By.XPath("//*[@class='header-search__submit']"));
+            searchClick.Click();
+
+            var wordToSearch = _chrome.Url.Replace("https://www.epam.com/search?q=", "");
+            Assert.That(wordToSearch, Is.EqualTo("Automation"));
+
+            var articleElements = _chrome
+                .FindElements(By.XPath("//*[@class='search-results__items']/article"))
+                .Take(5);
+
+            var articleElementsContent = articleElements.Select(x => x.Text);
+
+            Assert.True(articleElementsContent.All(x => x.Contains("Automation", StringComparison.OrdinalIgnoreCase)));
+        }
+
+        [Test]
+        public void CheckSearchResultLinkTextIsTheSameAsArticleHeader()
+        {
+            var action = new Actions(_chrome);
+
+            var searchButton = _chrome.FindElement(By.XPath("//*[@class='header-search__button header__icon']"));
+            searchButton.Click();
+
+            var searchPanelDisplayedState = Waiter.Until(function =>
+            {
+                try
+                {
+                    var searchPanel =
+                        _chrome.FindElement(By.XPath("//*[contains(@class, 'header-search__field no-focus')]"));
+                    return searchPanel.Displayed;
+                }
+                catch (NoSuchElementException)
+                {
+                    return false;
+                }
+            });
+            Assert.IsTrue(searchPanelDisplayedState);
+
+            var searchPanelClick = _chrome.FindElement(By.XPath("//*[@class='header-search__input frequent-searches__input']"));
+            searchPanelClick.Click();
+            searchPanelClick.SendKeys("Business Analysis");
+
+            var searchClick = _chrome.FindElement(By.XPath("//*[@class='header-search__submit']"));
+            searchClick.Click();
+
+            var wordToSearch = _chrome.Url.Replace("https://www.epam.com/search?q=", "");
+            Assert.That(wordToSearch, Is.EqualTo("Business+Analysis"));
+
+            var firstArticleElement = _chrome
+                 .FindElements(By.XPath("//*[@class='search-results__items']/article"))
+                 .First();
+
+            var linkElement = firstArticleElement.FindElement(By.XPath("//h3/a"));
+            var linkElementContent = linkElement.Text;
+            var linkElementUrl = linkElement.GetAttribute("href");
+
+            _chrome.Navigate().GoToUrl(linkElementUrl);
+
+            var articleHeader = _chrome.FindElement(By.XPath("//h1"));
+            var articleHeaderContent = articleHeader.Text;
+
+            Assert.That(linkElementContent, Is.EqualTo(articleHeaderContent));
```

Comment:
in this assert you check not text corrrectness but equality of link content and header. There could be a bug in this place and you will get false positive test